### PR TITLE
Automatically generate image tags

### PR DESCRIPTION
**Pull request by murphybytes**
See [Issue](https://app.shortcut.com/replicated/story/37104/automate-patching-container-image-cves).  Adds a utility that will discover the latest tag for images.  Two files are generated, a Go source file containing constant declarations, and a .env file containing environment variable image tag declarations.

### CODE REVIEW

File: cmd/imagedeps/main.go [hunk 1]
```diff
@@ -14,11 +14,15 @@ import (
 const (
 	minioReference                    = "minio"
 	dexReference                      = "dex"
+	postgresAlpineReference =  "postgres-alpine"
+	postgresDebianReference = "postgres-debian"
+	postgresAlpineTagFilter =  `^10.\d+-alpine$`
```
Comment (emosbaugh) [inline cmd/imagedeps/main.go]: These seem appropriate in the .image.env file as the third arg

File: .image.env [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Generated file, do not modify.  This file is generated from a text file containing a list of images. The
+# most recent tag is interpolated from the source repository and used to generate a fully qualified image
+# name.
```
Comment (DanStough) [inline .image.env:3]: Maybe a line here about how to regenerate or to a link in the readme.

File: pkg/image/constants.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package image
+
+// Generated file, do not modify.  This file is generated from a text file containing a list of images. The
+// most recent tag is interpolated from the source repository and used to generate a fully qualified
+// image name.
```
Comment (DanStough) [inline pkg/image/constants.go:5]: Same comment here.

File: cmd/imagedeps/main.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	parts := strings.Split(s, "-")
+	result := ""
+	for _, part := range parts {
+		result += strings.Title(part)
+	}
+	return result
+}
+
+// converts a name from the input file into an environment variable name
+// for example: foo_bar_baz -> FOO_BAR_BAZ
```
Comment (DanStough) [inline cmd/imagedeps/main.go:128]:
```suggestion
// for example: foo-bar-baz -> FOO_BAR_BAZ
```

**Review by DanStough — APPROVED** [review]

**Pull request merged**